crates/l2/based/sequencer_state.rs

```diff
@@ -0,0 +1,21 @@
+use std::sync::Arc;
+
+use tokio::sync::Mutex;
+
+pub type SequencerState = Arc<Mutex<SequencerStatus>>;
```

Comment:
We could wrap this and hide the fact it's a mutex by adding a getter/setter.

```suggestion
pub struct SequencerState(Arc<Mutex<SequencerStatus>>);

impl SequencerState {
    pub async fn get_status(&self) -> SequencerStatus {
        *self.clone().lock().await
    }

    pub async fn set_status(&self, new_status: SequencerStatus) {
        *self.clone().lock().await = new_status;
    }
}
```